### PR TITLE
docs: added example in docs of using a resolver for metatags for markdown component (#1014)

### LINKED ISSUE #1014: Docs: add example of using a resolver for metatags for markdown component
### Which scope/s are relevant/related to the feature request?

Docs

### Information

We have an example resolver for getting metatags for the route when using the custom markdown component.

https://github.com/analogjs/analog/blob/beta/apps%2Fblog-app%2Fsrc%2Fapp%2Fpages%2Fblog%2F%5Bslug%5D.page.ts#L16

### Describe any alternatives/workarounds you're currently using

_No response_

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by AmGarera**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/analogjs/analog/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [ ] vite-plugin-angular
- [ ] vite-plugin-nitro
- [ ] astro-angular
- [ ] create-analog
- [x] router
- [ ] platform
- [ ] content
- [ ] nx-plugin
- [ ] trpc

## What is the current behavior?

Closes #1014 

## What is the new behavior?

A new Example added to the docs.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: apps/docs-app/docs/features/routing/content.md [hunk 1]
```diff
@@ -161,6 +161,70 @@ export default class BlogPostComponent {
 }
 ```
 
+### Using A Resolver For Metatags
+
+The `postMetaResolver` function is responsible for fetching the meta tags for a post. This function should return an array of meta tags. Here's an example:
```
Comment (brandonroberts) [inline apps/docs-app/docs/features/routing/content.md:166]:
```suggestion
In your route configuration, you can use the `RouteMeta` object to resolve meta tags for a route. This is done by assigning the `postMetaResolver` function to the `meta` property. 

Below is an example of using a `postMetaResolver` function that fetches the meta tags for a post. This function returns an array of meta tags.
```

File: apps/docs-app/docs/features/routing/content.md [hunk 2]
```diff
@@ -161,6 +161,70 @@ export default class BlogPostComponent {
...
+  meta: postMetaResolver,
+};
+```
+
+The resolved meta tags can be accessed in the `BlogPostComponent` using the `ActivatedRoute` service. These meta tags can then be passed to the `MarkdownComponent`. Here's an example:
+
+```ts
+export default class BlogPostComponent {
+  readonly route = inject(ActivatedRoute);
+  readonly metaTags$ = this.route.data.pipe(pluck('meta'));
```
Comment (brandonroberts) [inline apps/docs-app/docs/features/routing/content.md:211]: Isn't `pluck` deprecated in RxJS?

```suggestion
  readonly metaTags$ = this.route.data.pipe(map(data => data['meta']));
```

File: apps/docs-app/docs/features/routing/content.md [hunk 3]
```diff
@@ -161,6 +161,70 @@ export default class BlogPostComponent {
...
+
+  // In the template
+  <analog-markdown [content]="post.content" [metaTags]="metaTags$ | async"></analog-markdown>
```
Comment (brandonroberts) [inline apps/docs-app/docs/features/routing/content.md:214]:
```suggestion
  <my-component [metaTags]="metaTags$ | async"></my-component>
```

File: apps/docs-app/docs/features/routing/content.md [hunk 4]
```diff
@@ -161,6 +161,70 @@ export default class BlogPostComponent {
...
+}
+```
+
+The `MarkdownComponent` should be modified to accept an input for meta tags and use these when rendering the markdown content. Here's an example:
```
Comment (brandonroberts) [inline apps/docs-app/docs/features/routing/content.md:218]: I think this section should be removed

**Review by brandonroberts — CHANGES_REQUESTED** [review]

**Comment by brandonroberts** [comment]
@AmGarera are you still interested in landing these changes?

**Comment by AmGarera** [comment]
Yeah I should get sometime later today to address your requested changes.

File: apps/docs-app/docs/features/routing/content.md (hunk 2 above)
Comment (AmGarera) [reply apps/docs-app/docs/features/routing/content.md:211]: Yup as of RxJS 7 it is

File: apps/docs-app/docs/features/routing/content.md (hunk 4 above)
Comment (AmGarera) [reply apps/docs-app/docs/features/routing/content.md:220]: Removed as requested

File: apps/docs-app/docs/features/routing/content.md [hunk 5]
```diff
@@ -169,7 +224,7 @@ Analog's markdown component supports [Mermaid](https://mermaid.js.org/). To enab
 withMarkdownRenderer({
   loadMermaid: () => import('mermaid'),
 });
-```
+````
```
Comment (brandonroberts) [inline apps/docs-app/docs/features/routing/content.md:227]:
```suggestion
```
```

**Review by brandonroberts — APPROVED** [review]

**Comment by brandonroberts** [comment]
Thanks! @allcontributors add @AmGarera for doc

**Pull request merged**